### PR TITLE
RAD-384 Use frontend-maven-plugin to fetch dependencies

### PR DESCRIPTION
**Pull request by Schwasrz94**
## Description
Setup the frontend-maven-plugin for the use of bower to fetch javascript, css and font dependencies and gulp to minify/uglify them.

* add frontend-maven-plugin to pom.xml
* create package.json to specify gulp dependencies
* create bower.json to specify radiology dependencies
* create gulpfile.js to automate copying and minifying of source files
* change includes according to new paths
* add node, node_modules, bower_components and vendor folder to gitignore
  file

## Related Issue

see https://issues.openmrs.org/browse/RAD-384

### CODE REVIEW

File: omod/pom.xml [hunk 1]
```diff
@@ -384,6 +384,47 @@
 					</images>
 				</configuration>
 			</plugin>
+			<plugin>
+				<groupId>com.github.eirslett</groupId>
+				<artifactId>frontend-maven-plugin</artifactId>
+				<version>1.0</version>
+				<executions>
+					<execution>
+					<id>install node and npm</id>
```
Comment (teleivo) [inline omod/pom.xml]: please indent

File: NOTICE [hunk 2]
```diff
@@ -55,4 +55,4 @@ Code License
     Applies to all CSS and LESS files in the following directories: font-awesome/css/, font-awesome/less/, and font-awesome/scss/.
     License: MIT License
     URL: http://opensource.org/licenses/mit-license.html
-Version : 4.6.3
+Version : 4.6.3
```
Comment (teleivo) [inline NOTICE]: you didnt change anything so please remove from diff

**Comment by teleivo** [comment]
@Schwasrz94 awesome :raised_hands: thank you!!

* can you please fix your build, see CI.
* try building the code on your branch and then checkout current master. youll see that git tells you that these folders are new and will be added

	bower_components/
	node/
	node_modules/
	src/main/webapp/resources/vendor/

we should do all that magic in the target dir which will be cleaned up by `mvn clean` and which prevents above issue. also it doesnt feel right to have the vendor dir put into the src/main/webapp/resources/ . in a js project you would also have the vendor in a build or bin dir and not in your source.

**Comment by Schwasrz94** [comment]
No problem ;)
That's why I added this folders to gitignore, but this doesn't worked out.. 
Wouldn't it be convenient to have the dependency sources more accessible than they are in the target dir?
Do I understand correctly that you would make a new build or bin folder inside the webapp dir with all vendor scripts? In my opinion they are resources and can obviously stay in the resource dir.

**Comment by teleivo** [comment]
what convenience would it give? I am not convinced ;)

I just dont want stuff thats generated via maven or any other tool to end up in our code dirs. It should end up in the target dir and be cleaned each time we run `mvn clean`. And if you do not want it to get cleaned (and skip doing the npm install over again) because you want to run tests again just dont add clean into the maven command.

**Comment by Schwasrz94** [comment]
If you want to look up some functions while coding you simply can look inside the bower_components folder :) 
But with all other stuff I totally agree. I will update the pr soon.

**Comment by teleivo** [comment]
@Schwasrz94 I have to apologize :( I played with it a little more and my point of putting node, node_modules, bower_components into the omod/target isnt good after all. I think this will cause most people to download the same stuff over and over again because theyll run clean. Can you change this back so that it ends up in omod/node, omod/node_modules, omod/bower_components (and add the folders to the .gitignore). Putting it all together should still happen in target and not in the src/main/webapp.

Other than that please add the few lines to the package.json.

The setup is very nice thanks :rocket: 
and very clean! Tried to update some packages and it just updates those. No nasty download and stick external deps source files into our repo ;) :+1: :+1:

**Comment by teleivo** [comment]
@Schwasrz94 oh and all the future things we can do now that we have proper tools, one can only dream (linting, minifying, testing our code), haha. thank you!

**Comment by Schwasrz94** [comment]
Done. But the build fails now again. What should we do about that?

**Comment by teleivo** [comment]
@Schwasrz94 exlcude the folders causing the errorfromthe license Plugin in the root pom. See CI log. Guess the node, node_modules, bower... Folders are the cause

**Comment by Schwasrz94** [comment]
@teleivo I excluded those folders also the first time after i opened the PR. I don't get it.. The license plugin even checks the headers before the node install, or am I wrong?

**Comment by teleivo** [comment]
@Schwasrz94 have you tried the exclude without putting omod/ in front and maybe also without **
the root pom config is inherited by API and omod. I think the paths are thus relative to api/ and omod/

**Comment by Schwasrz94** [comment]
@teleivo I tried without **, but not without omod. Good to know :)

**Comment by teleivo** [comment]
@Schwasrz94 :bowtie:

**Comment by Schwasrz94** [comment]
@teleivo Finally 😃

**Pull request merged**